post_processing_stages/tf_stage.cpp

```diff
@@ -0,0 +1,185 @@
+/* SPDX-License-Identifier: BSD-2-Clause */
+/*
+ * Copyright (C) 2021, Raspberry Pi (Trading) Limited
+ *
+ * tf_stage.hpp - base class for TensorFlowLite stages
+ */
+
+#include "tf_stage.hpp"
+
+TfStage::TfStage(LibcameraApp *app, int tf_w, int tf_h) : PostProcessingStage(app), tf_w_(tf_w), tf_h_(tf_h)
+{
+	if (tf_w_ <= 0 || tf_h_ <= 0)
+		throw std::runtime_error("TfStage: Bad TFLite input dimensions");
+}
+
+void TfStage::Read(boost::property_tree::ptree const &params)
+{
+	config_->number_of_threads = params.get<int>("number_of_threads", 2);
+	config_->refresh_rate = params.get<int>("refresh_rate", 5);
+	config_->model_file = params.get<std::string>("model_file", "");
+	config_->verbose = params.get<int>("verbose", 0);
+
+	initialise();
+
+	readExtras(params);
+}
+
+void TfStage::initialise()
+{
+	model_ = tflite::FlatBufferModel::BuildFromFile(config_->model_file.c_str());
+	std::cout << "TfStage: Loaded model " << config_->model_file << std::endl;
+	model_->error_reporter();
+
```

Comment:
If there is an error (e.g. model file missing), do we need some kind of exception here?

Reply:
Yes, I'll fix that.